poutyne/framework/iterators.py

```diff
@@ -44,6 +44,9 @@ def __init__(self, generator, steps_per_epoch, batch_metrics_names, callback=Non
             self.on_batch_begin = callback.on_test_batch_begin
             self.on_batch_end = callback.on_test_batch_end
 
+            callback.set_params({'steps': steps_per_epoch})
+            callback.on_test_begin({})
```

Comment:
Why do you remove this from Model to put it here? I think we should just add the set_params call at the same place as on_test_begin in Model.

Reply:
765a06d